src/dotnet/tests/Mil.Navy.Nrl.Norm.IntegrationTests/NormSessionTests.cs

```diff
@@ -0,0 +1,2139 @@
+
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            finally
+            {
+                StopSender();
+                StopReceiver();
+            }
+        }
+
+        [SkippableFact(typeof(IOException))]
+        public void EnqueuesData()
+        {
+            StartSender();
+            //Create data to write to the stream
+            var expectedContent = GenerateTextContent();
+            byte[] expectedData = Encoding.ASCII.GetBytes(expectedContent);
+
+            try
+            {
+                var normData = _normSession.DataEnqueue(expectedData, 0, expectedData.Length);
+                var expectedEventTypes = new List<NormEventType> { NormEventType.NORM_TX_OBJECT_SENT, NormEventType.NORM_TX_QUEUE_EMPTY };
+                var actualEventTypes = GetEvents().Select(e => e.Type).ToList();
+                Assert.Equal(expectedEventTypes, actualEventTypes);
+                var actualData = normData.Data;
+                Assert.Equal(expectedData, actualData);
+                var actualContent = Encoding.ASCII.GetString(actualData);
+                Assert.Equal(expectedContent, actualContent);
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            finally
+            {
+                StopSender();
+            }
+        }
+
+        [SkippableFact(typeof(IOException))]
+        public void ReceivesData()
+        {
+            _normSession.SetLoopback(true);
+            StartSender();
+            StartReceiver();
+
+            //Set up cache directory
+            var folderName = Guid.NewGuid().ToString();
+            var cachePath = Path.Combine(_testPath, folderName);
+            Directory.CreateDirectory(cachePath);
+            _normInstance.SetCacheDirectory(cachePath);
+
+            //Create data to be sent
+            var expectedContent = GenerateTextContent();
+            var expectedData = Encoding.ASCII.GetBytes(expectedContent);
+
+            try
+            {
+                var normData = _normSession.DataEnqueue(expectedData, 0, expectedData.Length);
+                var expectedEventTypes = new List<NormEventType>
+                {
+                    NormEventType.NORM_REMOTE_SENDER_NEW,
+                    NormEventType.NORM_REMOTE_SENDER_ACTIVE,
+                    NormEventType.NORM_TX_OBJECT_SENT,
+                    NormEventType.NORM_TX_QUEUE_EMPTY,
+                    NormEventType.NORM_RX_OBJECT_NEW,
+                    NormEventType.NORM_RX_OBJECT_UPDATED,
+                    NormEventType.NORM_RX_OBJECT_COMPLETED
+                };
+                var actualEvents = GetEvents();
+                AssertNormEvents(actualEvents);
+
+                var actualEventTypes = actualEvents.Select(e => e.Type).ToList();
+                Assert.Equivalent(expectedEventTypes, actualEventTypes);
+
+                var actualData = normData.Data;
```

Comment:
Shouldn't `actualData` come from the received events?

Reply:
Yes, it should use `actualEvents` to get `actualData`. I will fix the `ReceivesData()` test.